### PR TITLE
Tegra denver plat support v2

### PR DESCRIPTION
**Pull request by vwadekar**
Support for Tegra's "Project Denver" CPU based platforms

### CODE REVIEW

**Comment by vwadekar** [comment]
Previous PR - #339

**Comment by danh-arm** [comment]
Can one of the admins verify this patch?

**Comment by danh-arm** [comment]
test this please

File: docs/plat/nvidia-tegra.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ without changing any makefiles.
 Preparing the BL31 image to run on Tegra SoCs
 ===================================================
 CROSS_COMPILE=<path-to-aarch64-gcc>/bin/aarch64-none-elf- make PLAT=tegra \
```
Comment (athoelke) [inline docs/plat/nvidia-tegra.md]: This style doesn't work in markdown, the <> are interpreted as tags of some kind and the content is eliminated. You should blockquote this whole paragraph using ``` before and after, e.g.

```
CROSS_COMPILE=<path-to-aarch64-gcc>/bin/aarch64-none-elf- make PLAT=tegra \
TARGET_SOC=<target-soc e.g. t210|t132> BL32=<path-to-trusted-os-binary> \
SPD=<dispatcher e.g. tlkd> all
```

File: docs/plat/nvidia-tegra.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ without changing any makefiles.
 Preparing the BL31 image to run on Tegra SoCs
```
Comment (athoelke) [inline docs/plat/nvidia-tegra.md]: The document heading/title still only refers to T210

File: plat/nvidia/tegra/soc/t132/plat_psci_handlers.c [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+int tegra_prepare_cpu_suspend(unsigned int id, unsigned int afflvl)
+{
+	/* Nothing to be done for lower affinity levels */
+	if (afflvl < MPIDR_AFFLVL2)
+		return PSCI_E_SUCCESS;
+
+	/* Support for system suspend only */
+	if (id != PLAT_SYS_SUSPEND_STATE_ID) {
+		ERROR("%s: state-id not supported (%d)\n", __func__, id);
+		return PSCI_E_NOT_SUPPORTED;
```
Comment (athoelke) [inline plat/nvidia/tegra/soc/t132/plat_psci_handlers.c]: This error code is ignored by the caller - `tegra_affinst_suspend()` - and invalid `state_id` values cannot be tolerated at this stage in the suspend handling.

The framework does all parameter and state validation first before initiating changes that cannot be reversed. Errors must be reported only during the first phase via `plat_pm_ops::validate_power_state()`, then the framework assumes that `plat_pm_ops::affinst_suspend()` will always succeed.

File: plat/nvidia/tegra/soc/t132/plat_secondary.c [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+#define EVP_CPU_RESET_VECTOR		0x100
+
+extern void tegra_secure_entrypoint(void);
+
+/*
+ * For T132, CPUs reset to AARCH32, so the reset vector is first
+ * armv8_trampoline which does a warm reset to AARCH64 and starts
+ * execution at the address in SB_AA64_RESET_LOW/SB_AA64_RESET_HI.
+ */
+uint32_t __aligned(8) armv8_trampoline[] = {
```
Comment (athoelke) [inline plat/nvidia/tegra/soc/t132/plat_secondary.c]: Should be const

**Comment by athoelke** [comment]
This pull request has "checkpatch" errors. Please use "make checkpatch" and fix the resulting errors. You should also review the warnings but it is optional to fix those. Below is the checkpatch log from our build system for your information:

```
  CHECKING STYLE
WARNING: Do not use whitespace before Signed-off-by:
#12: 
    Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

WARNING: Do not use whitespace before Signed-off-by:
#42: 
    Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

WARNING: Duplicate signature
#42: 
    Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

WARNING: line over 80 characters
#90: FILE: plat/nvidia/tegra/common/tegra_sip_calls.c:67:
+	uint32_t spsr_64 = SPSR_64(MODE_EL2, MODE_SP_ELX, DISABLE_ALL_EXCEPTIONS);

WARNING: Possible switch case/default not preceeded by break or fallthrough comment
#109: FILE: plat/nvidia/tegra/common/tegra_sip_calls.c:103:
+	case TEGRA_SIP_AARCH_SWITCH:

WARNING: added, moved or deleted file(s), does MAINTAINERS need updating?
#135: 
new file mode 100644

WARNING: externs should be avoided in .c files
#392: FILE: plat/nvidia/tegra/soc/t132/plat_secondary.c:51:
+extern void tegra_secure_entrypoint(void);

ERROR: open brace '{' following function declarations go on the next line
#399: FILE: plat/nvidia/tegra/soc/t132/plat_secondary.c:58:
+uint32_t __aligned(8) armv8_trampoline[] = {

ERROR: do not use C99 // comments
#400: FILE: plat/nvidia/tegra/soc/t132/plat_secondary.c:59:
+	0xE3A00003,		// mov	r0, #3

ERROR: do not use C99 // comments
#401: FILE: plat/nvidia/tegra/soc/t132/plat_secondary.c:60:
+	0xEE0C0F50,		// mcr	p15, 0, r0, c12, c0, 2

ERROR: do not use C99 // comments
#402: FILE: plat/nvidia/tegra/soc/t132/plat_secondary.c:61:
+	0xEAFFFFFE,		// b	.

WARNING: line over 80 characters
#421: FILE: plat/nvidia/tegra/soc/t132/plat_secondary.c:80:
+	tegra_pmc_write_32(PMC_SECURE_SCRATCH22, (unsigned long)&armv8_trampoline);

WARNING: Do not use whitespace before Signed-off-by:
#593: 
    Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

WARNING: Duplicate signature
#593: 
    Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

WARNING: Do not use whitespace before Signed-off-by:
#824: 
    Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

WARNING: Duplicate signature
#824: 
    Signed-off-by: Varun Wadekar <vwadekar@nvidia.com>

total: 4 errors, 12 warnings, 705 lines checked

NOTE: Ignored message types: COMPLEX_MACRO GERRIT_CHANGE_ID GIT_COMMIT_ID

Your patch has style problems, please review.

If any of these errors are false positives, please report
them to the maintainer, see CHECKPATCH in MAINTAINERS.
```

**Comment by vwadekar** [comment]
New PR - #341

**Pull request closed without merging**